airbyte-integrations/connectors/source-salesforce/source_salesforce/streams.py

```diff
@@ -292,9 +292,9 @@ def download_data(self, url: str, chunk_size: float = 1024) -> os.PathLike:
         # set filepath for binary data from response
         tmp_file = os.path.realpath(os.path.basename(url))
         with closing(self._send_http_request("GET", f"{url}/results", stream=True)) as response:
-            with open(tmp_file, "w") as data_file:
+            with open(tmp_file, "wb") as data_file:
                 for chunk in response.iter_content(chunk_size=chunk_size):
-                    data_file.writelines(self.filter_null_bytes(self.decode(chunk)))
+                    data_file.write(self.filter_null_bytes(chunk))
```

Comment:
@Nakachi-S just worried that since the file was opened as `wb` and not `ab` the function `writelines()` that now needs to use `write()` which is what’s required for bytes might overwrite the file at each chunk. Other than that everything looks OK here for this particular file

Reply:
@koconder 
Thanks your review.

A temporary file is created for each URL, and chunks are continuously written to the file.
Therefore, it is not necessary to use "aw," which is an append mode.

Also about
> See feedback on how chunks are written, based on the code i believe only the last chunk will be written

 if you write "write" in the with clause as shown below, the result is not overwriting the file, but appending to it.

```
with open("tmp.txt", "wb") as f:
    f.wirte("Hello\n")
    f.write("World.)
```
this result is
```
Hello
World
```